tests/functional/test_completion.py

```diff
@@ -421,3 +421,28 @@ def test_completion_uses_same_executable_name(
         expect_stderr=deprecated_python,
     )
     assert executable_name in result.stdout
+
+
+@pytest.mark.parametrize(
+    "subcommand, handler_prefix, expected",
+    [
+        ("cache", "d", "dir"),
+        ("cache", "in", "info"),
+        ("cache", "l", "list"),
+        ("cache", "re", "remove"),
+        ("cache", "pu", "purge"),
+        ("config", "li", "list"),
+        ("config", "e", "edit"),
+        ("config", "ge", "get"),
+        ("config", "se", "set"),
+        ("config", "unse", "unset"),
+        ("config", "d", "debug"),
+        ("index", "ve", "versions"),
```

Comment:
this list could probably be auto-generated from all commands with a non-empty `handler_map` but I just hard coded it for simplicity (read: laziness :upside_down_face:)